Filter for currently running EC2 instances only

reedloden opened the pull request:
EC2 inventory should only contain actually running instances, not instances that may be in some other state (such as recently terminated). Add a filter to ensure that only instances with a current state of "running" are collected.

reedloden committed:
Filter for currently running EC2 instances only

EC2 inventory should only contain actually running instances, not instances
that may be in some other state (such as recently terminated). Add a filter
to ensure that only instances with a current state of "running" are collected.

itmecho commented:
Hi @reedloden, Thanks for the PR! I agree that some instances can be filtered out such as terminated, however I'm not sure we want to rule out stopped instances being included in the list. For example, if people have instances that are only running during certain hours, they would have to generate the inventory whilst they are running, otherwise they would end up with an incomplete inventory.

I think it might be better to add this a configuration flag, like `--ec2-only-get-running` (or something with a clear, logical name)

reedloden commented:
@itmecho how about we just set a sane default then? I'm fine with including "stopped" for this purpose. My main goal is to get rid of "terminated" instances.

Possible options: pending | running | shutting-down | terminated | stopping | stopped
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-lifecycle.html

How about including just: running | stopping | stopped? Would you be OK with that?

reedloden committed:
Add `stopping` and `stopped` instance states to filter

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-lifecycle.html

itmecho commented:
Looks good to me!
